### DISCUSSION TITLE
How to get the cell information on each circuit like the contents of the .pcap file?

### DISCUSSION THREAD

**Discussion by beenhuang** [shadow/shadow#1526] [General]
Hello,
In the shadow experimental environment, I want to get the cell information (including cell directions, sequence, cell type, inter-packet delay etc.) on all circuits of a specific relay, like the information of PCAP files. How should I do it?

Thank you,
Bin

**Comment by jtracey** [comment]
You can save PCAP files for individual hosts with the [`host_defaults.pcap_directory`](https://github.com/shadow/shadow/blob/c639d59d6edda581ccd5eac57044d45865af5aa4/docs/shadow_config_spec.md#host_defaultspcap_directory) setting.

**Reply by beenhuang** [reply]
I mean I want to get the cell information on individual circuits not packet information for individual hosts. How should I do?

**Comment by robgjansen** [comment]
The type of information you are looking for is specific to the Tor application, not to Shadow. You might find help on the Tor mailing lists.

As far as I know, the information you want is not directly available from the Tor log files, so I suppose you would need to first modify the Tor source code in order to retrieve it. You would need to read and understand the Tor code, and then add log statements to print out the information to stdout. Then you would build that version of Tor and run it in shadow, and the stdout file would then be captured during the shadow simulation.

Tor is quite complex so if you are not familiar with the Tor source code, then it might be rather difficult to modify the code and log the info you want. But it's not impossible and Tor is open source. Good luck :)

**Reply by beenhuang** [reply]
Thanks, I know what to do next.
